### PR TITLE
Create CLI Submitter

### PR DESCRIPTION
**Pull request by nbayindirli**
### Description

* A CLI-side submitter to interact with the SDK's tx submitter builder

### Drive-by changes

* None

### Related issues

* Fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3738

### Backward compatibility

* Yes

### Testing

* None

### CODE REVIEW

**Review by jmrossy — COMMENTED** [review]
Mostly looks good! Just a few small questions/comments to clear up.

I'd also appreciate it if we could exercise this in the `ci-test.sh` e2e test please. It's been useful for catching CLI regressions in the past.

File: typescript/cli/src/submit/submitTypes.ts
Comment (nbayindirli) [reply typescript/cli/src/submit/submitTypes.ts:1]: Ah– I thought I refactored this back to types. Updated

File: typescript/sdk/src/providers/transactions/submitter/TxSubmitterTypes.ts
Comment (nbayindirli) [reply typescript/sdk/src/providers/transactions/submitter/TxSubmitterTypes.ts:13]: I do not think so, since `safeAddress` is definitely only used to set the safeAddress for proposals

With that said, I updated the below field on `EV5ImpersonatedAccountTxSubmitterProps` to be `userAddress` for symmetry (also more accurate)

File: typescript/sdk/src/providers/transactions/transformer/ethersV5/EV5InterchainAccountTxTransformer.ts
Comment (yorhodes) [inline typescript/sdk/src/providers/transactions/transformer/ethersV5/EV5InterchainAccountTxTransformer.ts:47]: I suppose we should assert that the chain Id is consistent, or actually produce separate callRemote bundles for different chain Ids

**Review by yorhodes — COMMENTED** [review]
great start
I think we can cut down further on the number of new config interfaces here

**Review by jmrossy — APPROVED** [review]
✅ 
Recommend waiting for Yorke's approval too :)

File: typescript/cli/src/submit/submit.ts
Comment (yorhodes) [reply typescript/cli/src/submit/submit.ts]: this doesn't look resolved?

Comment (nbayindirli) [reply typescript/cli/src/submit/submit.ts]: > this doesn't look resolved?

We chatted about this in the office– since getTransformer must call getCallRemote (which in turn might make a call to network), we're ok with keeping this async

Comment (yorhodes) [reply typescript/cli/src/submit/submit.ts]: the async transform doesnt happen in this function iiuc

File: typescript/sdk/src/providers/transactions/submitter/TxSubmitterTypes.ts
Comment (yorhodes) [reply typescript/sdk/src/providers/transactions/submitter/TxSubmitterTypes.ts]: I know gnosis is ethereum specific
ethers v5 is a specific ethereum library and I was just wondering if this needs to be labeled with that

File: typescript/sdk/src/providers/transactions/submitter/ethersV5/EV5ImpersonatedAccountTxSubmitter.ts
Comment (yorhodes) [inline typescript/sdk/src/providers/transactions/submitter/ethersV5/EV5ImpersonatedAccountTxSubmitter.ts:36]: idk what the implication is of not doing this but should we 
`stopImpersonating` after submitting?

File: typescript/sdk/src/providers/transactions/submitter/ethersV5/EV5TxSubmitterTypes.ts
Comment (yorhodes) [inline typescript/sdk/src/providers/transactions/submitter/ethersV5/EV5TxSubmitterTypes.ts:13]: is chain needed for every submitter?
maybe we can omit that from each

**Review by yorhodes — APPROVED** [review]

File: typescript/sdk/src/providers/transactions/submitter/ethersV5/EV5ImpersonatedAccountTxSubmitter.ts
Comment (nbayindirli) [reply typescript/sdk/src/providers/transactions/submitter/ethersV5/EV5ImpersonatedAccountTxSubmitter.ts:36]: Interesting- I think we should do this in order to keep the submission atomic, and I don't think this should have any downstream effects since we are also impersonating during this call

will address as a part of https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3786

File: typescript/sdk/src/providers/transactions/submitter/ethersV5/EV5TxSubmitterTypes.ts
Comment (nbayindirli) [reply typescript/sdk/src/providers/transactions/submitter/ethersV5/EV5TxSubmitterTypes.ts:13]: not actually – `EV5ImpersonatedAccountTxSubmitterProps` does not actually need a specific network tied to it, so will remove chain here. Gnosis does need it in order to verify the submitted transactions can be used with that submitter

will address as a part of https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3786

**Pull request merged**
